pkg/virt-operator/resource/apply/delete.go

```diff
@@ -632,14 +631,15 @@ func crdHandleDeletion(kvkey string,
 
 	for _, crd := range needFinalizerAdded {
 		crdCopy := crd.DeepCopy()
-		controller.AddFinalizer(crdCopy, v1.VirtOperatorComponentFinalizer)
+		controller.AddFinalizer(crdCopy, v1.VirtOperatorComponentFinalizer)	
+
+		payload, err := patch.New(patch.WithAdd(finalizerPath, crdCopy.Finalizers)).GeneratePayload()
 
-		patchBytes, err := json.Marshal(crdCopy.Finalizers)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to generate patch payload: %v", err)
```

Comment:
Do we need to change the error message here?

Reply:
What should it be changed to? Technically it's multiple operations in one line but only `GeneratePayload()` returns an error

Reply:
Simply return err as in the previous code?

Reply:
Oh. I added the error message for consistency, line 537 which also uses `GeneratePayload()` has this error message, I also added it at line 678